src/saveload/saveload.h

```diff
@@ -326,7 +326,9 @@ enum SaveLoadVersion : uint16 {
 	SLV_INDUSTRY_TEXT,                      ///< 289  PR#8576 v1.11.0-RC1  Additional GS text for industries.
 	SLV_MAPGEN_SETTINGS_REVAMP,             ///< 290  PR#8891 v1.11  Revamp of some mapgen settings (snow coverage, desert coverage, heightmap height, custom terrain type).
 
-	SL_MAX_VERSION,                         ///< Highest possible saveload version
+	SL_MAX_VERSION,                         ///< Highest possible saveload version [for upstream versions until this one]
+
+	SLV_JGR_EXT_BASE = (1 << 15)            ///< 32768 JGRPP sets this bit to distinguish savegames using its extended versioning system.
```

Comment:
I think it is wrong to mention JGRPP here, as by what you pointed out yourself, this is the advise for all patchpack creators. Additionally, it is a bit-check, so maybe we should name it as such.

`SL_PATCHPACK_FLAG = 1 << 15 ///< Patchpacks should set this bit to distinguish savegames from vanilla`

or something?

Reply:
We should recommend that all new patchpacks use JGR's extended versioning system specifically; not blindly set this as an "I'm a patchpack" flag.

I thought that was what you meant in the original comment, but apparently not...

Will rephrase the comments, and see if @JGRennison has any docs we can link to.

In principle, we could also add support for loading such savegames into upstream versions, if they use only extended features marked as discardable/ignorable.

Reply:
No, you are right. I mostly meant the "JGRPP sets this bit .." when we want all patchpacks to do this. "Patchpacks set this bit.." makes more sense.

We still refer to the way JGRPP extended the savegame format, with feature toggles etc. I just never bothered to properly refer to that, so yeah, anything another patchpack author can pick up from would be greatly appreciated!

Reply:
The only documentation is here: https://github.com/JGRennison/OpenTTD-patches/blob/jgrpp/src/saveload/extended_ver_sl.cpp#L8-L36, which is admittedly a bit sparse.